tools/yul_phaser/Chromosome.h

```diff
@@ -0,0 +1,55 @@
+/*
+	This file is part of solidity.
+
+	solidity is free software: you can redistribute it and/or modify
+	it under the terms of the GNU General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	solidity is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+	GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with solidity.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include <map>
+#include <ostream>
+#include <string>
+#include <vector>
+
+namespace solidity::phaser
+{
+
+class Chromosome
+{
+public:
+	Chromosome():
+		m_optimizationSteps{} {}
+	explicit Chromosome(std::vector<std::string> const& _optimizationSteps):
+		m_optimizationSteps(_optimizationSteps) {}
+	explicit Chromosome(std::vector<std::string>&& _optimizationSteps):
+		m_optimizationSteps(_optimizationSteps) {}
+	static Chromosome make_random(int _length);
+
+	size_t length() const { return m_optimizationSteps.size(); }
+
+	std::vector<std::string> const& asSequence() const { return m_optimizationSteps; }
```

Comment:
```suggestion
	std::vector<std::string> const& stepNames() const { return m_optimizationSteps; }
```
or just 
```suggestion
	std::vector<std::string> const& optimizationSteps() const { return m_optimizationSteps; }
```
?

Reply:
I wanted to emphasize the fact that this method does not give you a copy of these steps but lets you use an existing object as if it were cast to a sequence of steps. Kinda like  `arguments["input-file"].as<string>()` in `boost::program_options`.

But well, you can look at it like an accessor too. I still prefer my name but I can change it if you think yours is clearer.

Reply:
OK, but my main question here would be: It returns this as a sequence, but a sequence of what? The fact that it does not return a copy is obvious from that it returns a `const&`.

Reply:
Right. This should be more specific.

Anyway, after using it a bit more I now agree with your initial suggestion. I do often need to use it as an accessor and `asSequence()` isn't too great in that context. Changed to `optimisationSteps()`.